### PR TITLE
Prevents being added twice to the list of Sintouched

### PR DESCRIPTION
**Pull request by Profakos**
The influenceSin() proc already adds the player to the list of people
who are sintouched. 

Fixed a check for already being sintouched in the influence sin spell , thanks @Cyberboss!

No known issue reports to close.

### CODE REVIEW

File: code/datums/mind.dm [hunk 1]
```diff
@@ -1172,7 +1172,6 @@
 				log_admin("[key_name(usr)] has devil'ed [current]. The devil has been marked as ascendable.")
 			if("sintouched")
 				if(ishuman(current))
-					ticker.mode.sintouched += src
```
Comment (Jalleo) [inline code/datums/mind.dm]: Uh this would only add if not already added I believe. Can someone confirm this?

Comment (Cyberboss) [inline code/datums/mind.dm]: Nope

Comment (Jalleo) [inline code/datums/mind.dm]: Okay then I mis remembered then but there is a way to only add if it isnt in there already correct?

Comment (Profakos) [inline code/datums/mind.dm]: If I am correct, |= would do that, but changing it on this line would not fix the error, I would have to change it in influenceSin() too. However, it seems to me that the convention of adding people to the antagonist ticker is by the += operator.

**Comment by Profakos** [comment]
While I am touching sins, I suppose i should also fix this line
https://github.com/tgstation/tgstation/blob/55b2b3461d5b00cfb9ff5c0660e3c32bd61e94eb/code/modules/spells/spell_types/devil.dm#L195
The continue only continues the inner loop, and not the outer loop, so it does nothing at all to prevent someone from being hit multiple times by the sintouch's effect.

**Comment by Cyberboss** [comment]
should be `if(locate(/datum/objective/sintouched) in H.mind.objectives`

**Comment by Profakos** [comment]
Thank you!

**Pull request merged**